File: rest-common/src/main/java/org/chronopolis/rest/listener/UpdatableEntityListener.java
package org.chronopolis.rest.listener;

import org.chronopolis.rest.entities.UpdatableEntity;

import javax.persistence.PrePersist;
import javax.persistence.PreUpdate;
import java.time.ZonedDateTime;

/**
 * EntityListener which updates the created_at on the initial save,
 * and updated_at on all others
 *
 * Created by shake on 3/21/16.
 */
public class UpdatableEntityListener {

    @PrePersist
    public void createTimeStamps(UpdatableEntity ue) {
        ue.setCreatedAt(ZonedDateTime.now());
        ue.setUpdatedAt(ZonedDateTime.now());
    }

    @PreUpdate
    public void updateTimeStamps(UpdatableEntity ue) {
        ue.setUpdatedAt(ZonedDateTime.now());
    }

}

File: rest-entities/src/main/kotlin/org/chronopolis/rest/kot/entities/listeners/UpdatableEntityListener.kt
package org.chronopolis.rest.kot.entities.listeners

import org.chronopolis.rest.kot.entities.UpdatableEntity
import java.time.ZoneOffset
import java.time.ZonedDateTime
import javax.persistence.PrePersist
import javax.persistence.PreUpdate

/**
 * EntityListener which updates the created_at on the initial save,
 * and updated_at on all others
 *
 * @author shake
 */
class UpdatableEntityListener {
    @PrePersist
    fun createTimeStamps(ue: UpdatableEntity) {
        println("HEY A PREPERSIST HERE")
        ue.createdAt = ZonedDateTime.now(ZoneOffset.UTC)
        ue.updatedAt = ZonedDateTime.now(ZoneOffset.UTC)
    }

    @PreUpdate
    fun updateTimeStamps(ue: UpdatableEntity) {
        println("HEY A UPDATING HERE")
        ue.updatedAt = ZonedDateTime.now(ZoneOffset.UTC)
    }
}